tests/integration/test_app.py

```diff
@@ -0,0 +1,42 @@
+# Standard Library
+from time import sleep
+
+# Third Party
+from requests import get, post
+
+
+def test_documents_endpoints():
+    """ Tests calling '/documents' endpoint with PDF file and then '/documents/{document_id}'
```

Comment:
this kind of tests are really heavy, as they need an actual instance of the service. I'd personally rather use the `fastapi.TestClient` (for more details https://fastapi.tiangolo.com/tutorial/testing/) that provides only a little bit lower level of integration but should be much faster and would not require a running instance of the service.

Reply:
I am aware of the `TestClient`, we used that quite often and if I would do the full test coverage I would definetely use it. The purpose of this test and the package of testing is really test the whole flow end to end... and that also means to spin up images with postgres, rabitmq and dramatiq. Without that (and this is actually in one of the unit test) I would need to mock all the dependecies. In real / bigger project containing multiple microservices those kind of tests would be definitely out of the repo as it would test the end to end flow cross different microservices, cross different repos, however for sake of testing this small project I decided to keep it there to have a easy way how to test it all together. If it would slow down the CI/CD significantly then I can imagine having this kind of test as a part of nightly builds.

Reply:
Cool